sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py

```diff
@@ -3,24 +3,53 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 # ------------------------------------
-
 import warnings
+from datetime import datetime
 from enum import Enum
-from typing import TYPE_CHECKING, Dict, Any, List
+from typing import Dict, Any, List
 
 from ._generated.models import (
     ArtifactTagProperties as GeneratedArtifactTagProperties,
     ContainerRepositoryProperties as GeneratedRepositoryProperties,
     RepositoryWriteableProperties,
     TagWriteableProperties,
     ManifestWriteableProperties,
+    ManifestAttributesBase,
 )
 from ._helpers import _host_only, _is_tag, _strip_alg
 
-if TYPE_CHECKING:
-    from typing import IO
-    from datetime import datetime
-    from ._generated.models import ManifestAttributesBase, OCIManifest
+
+class ArtifactArchitecture(str, Enum): # pylint: disable=enum-must-inherit-case-insensitive-enum-meta
```

Comment:
These have been added? They should inherit from the core case-insensitive enum meta class.

Reply:
Ah I see - there were moved up from the bottom of the file. We should still fix the base class.